### PR TITLE
13387 deployment scenario improvements

### LINKED ISSUE #13387: [FEATURE] Improving the "Deployment scenario" section of the user guide
### Problem description

This is a tracking and feedback issue for a documentation PR that I'm planning to submit. 

Currently, the "Deployment scenario" section of the user guide does not cover the `server_document` and `server_session` methods used to embed Bokeh server views into other web applications.

Additionally, my naive understanding is that BokehJS currently does not forward Authorization headers when sending the initial HTTP(S) request to set up its websocket connection to Bokeh server. If this is correct, it would be extremely helpful to note that Bokeh server's websocket endpoint must be exempted by auth modules implemented per the "Authentication" section of the "Deployment scenario" page. (I'm not much of a frontend dev and I have limited experience with websockets - apologies if I'm off-base here.)

### Feature description

I have written a documentation subsection covering the two embedding methods mentioned above.

If anyone has bandwidth or past experience to confirm the websocket + auth module issue in the next few days, I will include a few lines regarding that limitation in the PR linked to this issue - otherwise, I'll set it aside for now. The most relevant sections of the Bokeh code are under `bokehjs/src/lib/client`, which appear to call out to this websocket library: https://github.com/websockets/ws/.

### Potential alternatives

n/a

### Additional information

Original discourse thread here: https://discourse.bokeh.org/t/improving-the-bokeh-server-deployment-scenario-section-of-the-user-guide/10852.

### PR DESCRIPTION
**Pull request by wswope-etiometry**
This PR adds a new section to the "Deployment scenario" section of the user guide, adding detail about configuring Bokeh server for integration into other web applications, and linking out to other relevant sections of the docs.

All feedback & suggestions for revision are welcome.

- [ ] issues: fixes #13387
- [ ] tests added / passed: Passing all pre-commit hooks

### CODE REVIEW

**Comment by bryevdv** [comment]
@tcmetzger if you have a chance, please review this small docs PR 🙏

File: docs/bokeh/source/docs/user_guide/server/deploy.rst [hunk 1]
```diff
@@ -17,6 +17,52 @@ requirements and the expected number of users.
...
+If the parent service you wish to integrate with is not Python-based, you can
+still integrate with Bokeh through the ``server_document`` / ``server_session``
+methods, but you will need to do so by calling out to a small, long-lived
+Python script that returns the text contents of those methods via any standard
+form of IPC. Alternatively, you can copy the static contents of the autoloader
+script into your parent app directly as a string, and change the script tag's
+id attribute yourself. However, this approach is not recommended, as it will
+require manually validating the script text stored in your parent app as you
+upgrade across Bokeh versions, and may break entirely across major versions if
+the autoloader workflow is changed.
```
Comment (bryevdv) [inline docs/bokeh/source/docs/user_guide/server/deploy.rst]: @wswope-etiometry I think maybe we could just dispense with this "Alternatively, ..." I don't really want to appear to sanction this approach, even half-heartedly.

File: docs/bokeh/source/docs/user_guide/server/deploy.rst [hunk 2]
```diff
@@ -17,6 +17,52 @@ requirements and the expected number of users.
...
+dashboards and seamlessly embedding them within a larger parent application.
+To accomplish this, the ``bokeh.embed`` module offers the
+:func:`~bokeh.embed.server_document` and :func:`~bokeh.embed.server_session`
+methods. See :ref:`ug_output_embed_apps` for a detailed discussion of their
+usage, with examples.
+
+If the parent service you wish to integrate with is not Python-based, you can
+still integrate with Bokeh through the ``server_document`` / ``server_session``
+methods, but you will need to do so by calling out to a small, long-lived
+Python script that returns the text contents of those methods via any standard
```
Comment (bryevdv) [inline docs/bokeh/source/docs/user_guide/server/deploy.rst:40]: I don't really understand this scenario fully, would it be possible to give an abbreviated but concrete example? i.e. "parent service" could mean very many things but I am not sure exactly what it refers to here.

File: docs/bokeh/source/docs/user_guide/server/deploy.rst [hunk 3]
```diff
@@ -17,6 +17,52 @@ requirements and the expected number of users.
 However, if you have authentication, scaling, or uptime requirements, you'll
 have to consider more sophisticated deployment configurations.
 
+Integrating Bokeh server into other web services
+------------------------------------------------
+
+Bokeh server commonly acts as a solution for constructing plots and
+dashboards and seamlessly embedding them within a larger parent application.
```
Comment (bryevdv) [inline docs/bokeh/source/docs/user_guide/server/deploy.rst]:
```suggestion
dashboards and embedding them within a larger parent application.
```
no need for marketing speak here :)

File: docs/bokeh/source/docs/user_guide/server/deploy.rst [hunk 4]
```diff
@@ -17,6 +17,52 @@ requirements and the expected number of users.
...
+methods, but you will need to do so by calling out to a small, long-lived
+Python script that returns the text contents of those methods via any standard
+form of IPC. Alternatively, you can copy the static contents of the autoloader
+script into your parent app directly as a string, and change the script tag's
+id attribute yourself. However, this approach is not recommended, as it will
+require manually validating the script text stored in your parent app as you
+upgrade across Bokeh versions, and may break entirely across major versions if
+the autoloader workflow is changed.
+
+You will need to add the public hostname of your Bokeh server instance to the
```
Comment (bryevdv) [inline docs/bokeh/source/docs/user_guide/server/deploy.rst]: I think this sentence paragraph needs some motivation, i.e. should start like "In order to <accomplish the thing>, you will need to ..." what is "the thing" in this case?

Comment (bryevdv) [reply docs/bokeh/source/docs/user_guide/server/deploy.rst]: Maybe actually a comment for the top of the section to. IIRC this section is really to offer advice for the needs *specifically* around CSP configuration, maybe that could be stated from the start.

**Comment by wswope-etiometry** [comment]
Thanks for the feedback, @bryevdv! I agree with all your points.

Revised version has been pushed.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @wswope-etiometry everything looks good. There seems to be an unrelated issue caused by a new minor version of MyPy just getting released. I'll wait until we sort that out and then merge!

**Comment by bryevdv** [comment]
Apologies for the delay @wswope-etiometry this fell off my radar

**Pull request merged**